sthgetter/sthgetter.go

```diff
@@ -32,40 +33,40 @@ var logStr = "STH Getter"
 // Run runs an STH Getter, which periodically gets an STH from a Log, checks
 // that each one meets per-STH requirements defined in RFC 6962, and stores
 // them.
-func Run(ctx context.Context, lc *client.LogClient, st storage.APICallWriter, url string, period time.Duration) {
-	log.Printf("%s: %s: started with period %v", url, logStr, period)
+func Run(ctx context.Context, lc *client.LogClient, st storage.APICallWriter, l *ctlog.Log, period time.Duration) {
+	log.Printf("%s: %s: started with period %v", l.URL, logStr, period)
 
 	t := time.NewTicker(period)
 	defer t.Stop()
 	for {
 		select {
 		case <-t.C:
 			// TODO(katjoyce): Work out when and where to add context timeouts.
-			getCheckStoreSTH(ctx, url, lc, st)
+			getCheckStoreSTH(ctx, lc, st, l)
 		case <-ctx.Done():
-			log.Printf("%s: %s: stopped", url, logStr)
+			log.Printf("%s: %s: stopped", l.URL, logStr)
 			return
 		}
 
 	}
 }
 
-func getCheckStoreSTH(ctx context.Context, url string, lc *client.LogClient, st storage.APICallWriter) {
+func getCheckStoreSTH(ctx context.Context, lc *client.LogClient, st storage.APICallWriter, l *ctlog.Log) {
 	// Get STH from Log.
-	log.Printf("%s: %s: getting STH...", url, logStr)
+	log.Printf("%s: %s: getting STH...", l.URL, logStr)
 	_, httpData, getErr := lc.GetSTH()
 	if getErr != nil {
-		log.Printf("%s: %s: error getting STH: %s", url, logStr, getErr)
+		log.Printf("%s: %s: error getting STH: %s", l.URL, logStr, getErr)
 	}
 	if len(httpData.Body) > 0 {
-		log.Printf("%s: %s: response: %s", url, logStr, httpData.Body)
+		log.Printf("%s: %s: response: %s", l.URL, logStr, httpData.Body)
 	}
 
 	// Store get-sth API call.
 	apiCall := apicall.New(ct.GetSTHStr, httpData, getErr)
-	log.Printf("%s: %s: writing API Call...", url, logStr)
-	if err := st.WriteAPICall(ctx, apiCall); err != nil {
-		log.Printf("%s: %s: error writing API Call %s: %s", url, logStr, apiCall, err)
+	log.Printf("%s: %s: writing API Call...", l.URL, logStr)
```

Comment:
Not necessarily for this review but are you clear on where the write transaction boundaries are in the design?